samcli/lib/cookiecutter/processor.py

```diff
@@ -0,0 +1,22 @@
+""" Define a processor to process the cookiecutter context before/after generating a cookiecutter project"""
+from abc import ABC
+from typing import Dict
+
+
+class Processor(ABC):
+    """
+    An abstract class for defining template's preprocessors and postprocessors
+    """
+
+    def run(self, context: Dict):
+        """
+        the processing logic of this processor
+
+        Parameters
+        ----------
+        context: Dict
+            The cookiecutter context to process
+
+        Returns: A processed copy of the cookiecutter context
+        """
+        raise NotImplementedError("not implemented")
```

Comment:
Adding a `@abstractmethod` instead of raise an exception here? https://docs.python.org/3/library/abc.html and https://softwareengineering.stackexchange.com/questions/326207/using-notimplementederror-instead-of-abstract-classes